src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployAllTask.java

```diff
@@ -36,17 +39,30 @@ public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFact
     this.cloudSdkBuilderFactory = cloudSdkBuilderFactory;
   }
 
+  public void setStageDirectory(File stageDirectory) {
+    this.stageDirectory = stageDirectory;
+  }
+
   /** Task Entrypoint : Deploys the app and all of its config files. */
   @TaskAction
   public void deployAllAction() throws AppEngineException {
-    if (!deployConfig.getDeployables().isEmpty()) {
-      getLogger().warn("appengineDeployAll: Ignoring configured deployables.");
-      deployConfig.getDeployables().clear();
+    List<File> deployables = deployConfig.getDeployables();
+
+    // Clear out configured deployables
+    if (!deployables.isEmpty()) {
+      deployables.clear();
+    }
+
+    // Look for app.yaml
+    File appYaml = new File(stageDirectory, "app.yaml");
+    if (!appYaml.exists()) {
+      throw new GradleException("Failed to deploy all: app.yaml not found.");
     }
+    getLogger().info("appengineDeployAll: Preparing to deploy app.yaml");
```

Comment:
This could be combined with the same log message below in a separate method like you did in the `app-maven-plugin`.

An alternative way to do this would be to structure the flow as:

1. Start list of empty yaml files.
2. Add `app.yaml` file or give exception.
3. Add other yaml files.
4. Check, log, and add files to `deployables`.